src/sgame/sg_weapon.cpp

```diff
@@ -46,18 +46,32 @@ static bool TakesDamages( gentity_t const* ent )
 
 void G_ForceWeaponChange( gentity_t *ent, weapon_t weapon )
 {
+	if ( G_Team( ent ) != TEAM_HUMANS ) // only humans have several weapons
+	{
+		return;
+	}
 	playerState_t *ps = &ent->client->ps;
 
+	weapon_t prevWeapon = static_cast<weapon_t>( ps->persistant[ PERS_NEWWEAPON ] );
```

Comment:
I believe you should use BG_GetPlayerWeapon to get the prev weapon

Reply:
I used this originally, and results felt really bogus. But maybe this was because of other changes, I don't remember for certain (there's more than a few improvements in that PR).

Reply:
The new weapon is set to 0 upon spawning so it will be wrong then at the very least.